src/main/java/sg/dex/starfish/impl/squid/DexResolver.java

```diff
@@ -0,0 +1,128 @@
+import com.oceanprotocol.keeper.contracts.DIDRegistry;
+
+public class DexResolver implements Resolver {
+    private DIDRegistry contract;
+    private SquidService squidService;
+
+    /**
+     * Create DexResolver
+     *
+     * @param DIDRegistry contract
+     * @param SquidService squidService
+     */
+    private DexResolver(DIDRegistry contract, SquidService squidService)  {
+        this.contract = contract;
+        this.squidService = squidService;
+    }
+
+    /**
+     * Creates a SquidResolverImpl
+     *
+     * @param String configFile. All information about account credentials will be taken from this file
+     * @throws IOException, CipherException
+     * @return SquidResolverImpl The newly created SquidResolverImpl
+     */
+    public static DexResolver create(String configFile) throws IOException, CipherException{
+        SquidService squidService = SquidService.create(configFile);
+        Properties properties = squidService.getProperties();
+        String address = (String)properties.getOrDefault("contract.DIDRegistry.address", "");
+        OceanConfig oceanConfig = OceanConfigFactory.getOceanConfig(properties);
+        KeeperService keeper = squidService.getKeeperService(oceanConfig);
+        DIDRegistry contract = DIDRegistry.load(address, keeper.getWeb3(), keeper.getTxManager(), keeper.getContractGasProvider());
+        return new DexResolver(contract, squidService);
+    }
+
+    @Override
+    public String getDDOString(DID did) {
+        com.oceanprotocol.squid.models.DID squidDID = null;
+        try {
+            squidDID = new com.oceanprotocol.squid.models.DID(did.toString());
+        } catch (DIDFormatException e) {
+            e.printStackTrace();
+            return null;
+        }
+
+        String didHash = squidDID.getHash();
+        BigInteger blockNumber = BigInteger.valueOf(0);
+        try {
+            blockNumber = (BigInteger) contract.getBlockNumberUpdated(EncodingHelper.hexStringToBytes(didHash)).send();
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+            throw Utils.sneakyThrow(e);
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw Utils.sneakyThrow(e);
+        }
+
+        EthFilter filter = new EthFilter(DefaultBlockParameter.valueOf(blockNumber), DefaultBlockParameter.valueOf(blockNumber), contract.getContractAddress());
+        filter.addSingleTopic(EventEncoder.encode(contract.DIDATTRIBUTEREGISTERED_EVENT));
+        String didTopic = "0x" + didHash;
+        filter.addOptionalTopics(new String[]{didTopic});
+
+        Flowable<DIDRegistry.DIDAttributeRegisteredEventResponse> floable = contract.dIDAttributeRegisteredEventFlowable(filter);
+        ArrayList<DIDRegistry.DIDAttributeRegisteredEventResponse> outcome = new ArrayList<>();
+        floable.subscribe(log -> {
+            outcome.add(log);
+        });
+
+        return outcome.size() == 1 ? outcome.get(0)._value : null;
+    }
+
```

Comment:
Should be some tests for Agent registration / lookup, i.e. I want to register an agent on the resolver, then get the same agent back by looking up via DID